src/main/java/org/wildfly/security/credential/external/ExternalCredential.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.credential.external;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.credential.Credential;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * External credential.
+ * It is source for credentials obtained from external source.
+ *
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>
+ */
+public class ExternalCredential {
```

Comment:
Why not just use CallbackHandlers in the opposite way, the way they're used elsewhere: accept a callback hander from the user which uses the existing NameCallback and CredentialCallback to acquire a credential by name from the underlying system?  Many of these classes are essentially duplicates of classes which already exist in Elytron, with just slightly different shape.  See below...